include/hal_core/utilities/enums.h

```diff
@@ -10,58 +10,60 @@ namespace hal
      * Container storing the string description of an enum.
      * Must be declared for every enum to be supported.
      */
-    template<typename T>
+    template<typename T, typename = typename std::enable_if<std::is_enum<T>::value, T>::type>
     struct EnumStrings
     {
         static std::vector<std::string> data;
     };
 
     /**
      * Translates an enum value into its string representation.
-     * 
+     *
      * @param[in] e - The enum value.
      * @returns The string representation.
      */
-    template<typename T>
-    std::string enum_to_string(T const& e)
+    template<typename T, typename = typename std::enable_if<std::is_enum<T>::value, T>::type>
+    std::string enum_to_string(T e)
     {
-        return EnumStrings<T>::data.at(static_cast<int>(e));
```

Comment:
exception for vector out of bounds access when no string for the enum value is present